aosp_diff/preliminary/external/drm_hwcomposer/0006-fix-static-bound-to-dynamically-bound.patch

```diff
@@ -0,0 +1,229 @@
+From 866264bad61c61ddb7c092fb63fee120985559f1 Mon Sep 17 00:00:00 2001
+From: Xiao HaowenX <haowenx.xiao@intel.com>
+Date: Mon, 15 Nov 2021 13:33:49 +0800
+Subject: [PATCH] Previously connectors are bound to crtc in init stage while
+ connector counter and crtc counter are not matched then caused some displays
+ can't display when hotplug monitors into unbound connectors. Change to bind
+ connector and crtc dynamically when detect connector changed to CONNECTED
+ state in hotplug handling
+
+Tracked-On:OAM-99492
+Signed-off-by: Xiao HaowenX <haowenx.xiao@intel.com>
+---
+ DrmHwcTwo.cpp         | 39 ++++++++++++++++++++++++++++++++++-----
+ DrmHwcTwo.h           |  1 +
+ drm/DrmDevice.cpp     | 43 +++++--------------------------------------
+ drm/DrmDevice.h       |  9 ++++++---
+ drm/ResourceManager.h |  3 +++
+ 5 files changed, 49 insertions(+), 46 deletions(-)
+
+diff --git a/DrmHwcTwo.cpp b/DrmHwcTwo.cpp
+index 194b11e..bb474eb 100644
+--- a/DrmHwcTwo.cpp
++++ b/DrmHwcTwo.cpp
+@@ -1202,23 +1202,49 @@ void DrmHwcTwo::HandleInitialHotplugState(DrmDevice *drmDevice) {
+   }
+ }
+ 
++int DrmHwcTwo::DynamicallyBound(DrmConnector *mconn,DrmDevice *mdrm){
++	int curr_num_display = resource_manager_.getDisplayCount();
++	mconn->set_display(curr_num_display);
++	if (mdrm->max_pipe_count-- <= 0)
```

Comment:
we see max_pipe_count is decreased, then when hotplug out an monitor, we need increased the max_pipe_count. Also suggest to change max_pipe_count to free_pipe_count due the variable is changed dynamically.